components/ProductListing/ProductListingSorting.vue

```diff
@@ -0,0 +1,55 @@
+<script setup lang="ts">
+import type { Schemas } from '@shopware/api-client/api-types';
+
+const props = defineProps<{
+    options: Schemas['ProductListingResult']['availableSortings'];
+    selectedOption: Schemas['ProductListingResult']['sorting'];
+}>();
+const emit = defineEmits<{
+    'sorting-changed': [key: Schemas['ProductListingResult']['sorting']];
+}>();
+
+const toggleState = ref(false);
+
+const handleClick = (option: Schemas['ProductListingResult']['availableSortings'][0]) => {
+    toggleState.value = false;
+    emit('sorting-changed', option.key);
+};
+</script>
+
+<template>
+    <DropdownMenuRoot v-model:open="toggleState">
+        <DropdownMenuTrigger
+            class="flex items-center justify-between py-2 text-black"
+            :aria-label="$t('listing.sidebar.sorting.ariaLabel')"
+        >
+            <span class="text-left">{{ $t('listing.sidebar.sorting.text') }}</span>
+            <FormKitIcon
+                :icon="toggleState ? 'chevron-up' : 'chevron-down'"
+                class="ml-6 h-4 w-auto"
+            />
+        </DropdownMenuTrigger>
+        <DropdownMenuPortal>
+            <DropdownMenuContent
+                class="data-[side=top]:animate-slideDownAndFade data-[side=right]:animate-slideLeftAndFade data-[side=bottom]:animate-slideUpAndFade data-[side=left]:animate-slideRightAndFade min-w-[220px] rounded-md border border-gray bg-white p-[5px] shadow-[0px_10px_38px_-10px_rgba(22,_23,_24,_0.35),_0px_10px_20px_-15px_rgba(22,_23,_24,_0.2)] outline-none will-change-[opacity,transform]"
```

Comment:
```suggestion
                class="data-[side=top]:animate-slideDownAndFade data-[side=right]:animate-slideLeftAndFade data-[side=bottom]:animate-slideUpAndFade data-[side=left]:animate-slideRightAndFade min-w-[220px] rounded-md border border-gray bg-white p-1 shadow-md outline-none will-change-[opacity,transform]"
```
we try to avoid arbitrary values when we have similar ones already defined and spare using arbitrary values only for special cases (like will-change would be fine for me)